exporter/opentelemetry-exporter-otlp/src/opentelemetry/exporter/otlp/exporter.py

```diff
@@ -121,24 +134,40 @@ class OTLPExporterMixin(
 
     Args:
         endpoint: OpenTelemetry Collector receiver endpoint
+        insecure: Connection type
         credentials: ChannelCredentials object for server authentication
         metadata: Metadata to send when exporting
     """
 
     def __init__(
         self,
-        endpoint: str = "localhost:55680",
+        endpoint: Optional[str] = None,
+        insecure: Optional[bool] = None,
         credentials: ChannelCredentials = None,
         metadata: Optional[Tuple[Any]] = None,
     ):
         super().__init__()
 
-        self._metadata = metadata
+        endpoint = (
+            endpoint
+            or Configuration().EXPORTER_OTLP_ENDPOINT
+            or "localhost:55680"
+        )
+
+        if insecure is None:
```

Comment:
Nit: Why is the logic not the as for `endpoint`?

Reply:
I think we want to have this precedence: `insecure arg` > `Trace/Metric exporter specific env var` > `Generic env var`. Because `insecure` is a boolean it'd be tricky to use chained `or`.